### ISSUE TITLE
Handle client stale check results when satellite cannot connect to master

### ISSUE THREAD

**Issue by marcofl** [Icinga/icinga2#5381]
## Expected Behavior
When a satellite is not able connect to the master zone API anymore there should be an indicator that the clients which are connected via the satellite are UNKNOWN for example (from cluster_zone check). 

## Current Behavior
Instead when a satellite is not able connect to the master zone API anymore all check results for the clients, which are connected to that satellite, are stale from the masters perspective. you see the "Check results are late" in the icingaweb2. 

## Steps to Reproduce (for bugs)
1. have a MASTER < SAT < Client setup
2. have all checks green
3. do something like `iptables -A OUTPUT -p tcp --dport 5665 -j DROP` on the satellite, so it can no longer connect to the Master Zone
4. wait and see all check results being late in icingaweb2 on the master

## Context
With a lot of satellites and clients connected to them, this can prevent issues from being seen / alerted because it looks like only one host (the satellite) is down but actually It could be 100 hosts behind that satellite which we think of are okay, but in fact they could be totally down too - or we just don't know. 

I know I could craft some host dependencies, which check if the satellite is up. but as we have a lot of satellites with a lot of clients (600+) this would mean a lot of magic in the config. 

In my opinion, icinga2 knows how the cluster "tree" looks and should be able to somehow mark clients as "unknown" if the satellite zone is disconnected. 

## Your Environment

* Version used (`icinga2 --version`): 2.6.3-1
* Operating System and version: Ubuntu 14.04 / 16.04
* Enabled features (`icinga2 feature list`): api checker command ido-mysql mainlog notification

**Comment by dnsmichi** [comment]
This has been discussed over and over in issues, mailing lists, forum. Once again: Imagine the following check result time lines:

- Satellite continues to receive check results from clients, forwards them to the master
- Master receives those check results and dumps them into backends (IDO, Graphite, etc.)

1.) At some point the connection between the master and the satellite drops. 

- Satellite continues to do checks
- Satellite cannot send check results back to master
- Satellite stores them into the replay log, keeping the history intact once connection is re-established
- Master does not receive anything, check results are late

2.) Once connection is re-established again, the following happens:

- Satellite replays the check result history to the master
- Master processes old check results in line, and keeps your historical backends uptodate (i.e. for SLA reports, or metrics)

The problem is with changing the check result history on the master. Say you've turned all late check results into UNKNOWN before 2.) happens. 

- Master sets all late check results to UNKNOWN, which in turn means "new check results with timestamp xy"

2a.) Master check result history has been modified, now the connection is re-established.

- Satellite replays the check result history to the master
- Master drops received check results because there already is one (UNKNOWN)
- Your backends don't receive any historical data from the satellite

Basically you have a split brain scenario for historical checks. 

In order to get notifications about disconnected zones, there's the cluster-zone check. This also implies that check results are not received or late. This also is a single point of failure where notifications make sense, instead of having all check results in an unknown state for hosts/services belonging to a child zone.

If you don't think that notifications for the cluster-zone check are sufficient, you might add your own cluster check using the functions described inside the troubleshooting guides.

**Comment by marcofl** [comment]
@dnsmichi thank you for the good explanation. I agree. The only problem I see, is that when a satellite is disconnected we don't actually know if it's just one host or one satellite with 100 clients. 

but I fully agree with the historical data scenario you showed. I guess we'll just add something like "SAT:" to the display name of satellites, so we know this host is "more important" when it's down. 
or have some kind of overall "late check results" check. 

thank you.

**Comment by Ashoury** [comment]
Your answer is clear but i am monitoring data center that has 100 client we will get power outage and satellite goes down
How I can know I have power outage that means all 100 nods go down or just satellite ?
Is there any solution for that?
